security/src/main/java/io/micronaut/security/authentication/BlockingAuthenticationProviderFactory.java

```diff
@@ -57,21 +59,14 @@ <T> AuthenticationProvider<T> createAuthenticationProvider(BlockingAuthenticatio
         return new BlockingAuthenticationProviderAdapter<>(blockingAuthenticationProvider, scheduler);
     }
 
-    private static class BlockingAuthenticationProviderAdapter<T> implements AuthenticationProvider<T> {
-
-        private final BlockingAuthenticationProvider<T> blockingAuthenticationProvider;
-
-        private final Scheduler scheduler;
-
-        private BlockingAuthenticationProviderAdapter(BlockingAuthenticationProvider<T> blockingAuthenticationProvider, Scheduler scheduler) {
-            this.blockingAuthenticationProvider = blockingAuthenticationProvider;
-            this.scheduler = scheduler;
-        }
+    private record BlockingAuthenticationProviderAdapter<T>(
```

Comment:
I considered doing the same (making the class a record) since IntelliJ was suggesting it, but to me this isn't the use case for which records are intended. It's not really just a data carrier, and I don't think it needs the accessor methods that the record provides, etc.

Do you have another reason that I'm not thinking of?

Reply:
I have reverted it https://github.com/micronaut-projects/micronaut-security/pull/1532/commits/03ddaf3b34722ba8c4687d71bc2cdc4385058792